vulcan/lib/client/jsx/components/workflow/session/advanced_session_controls.tsx

```diff
@@ -0,0 +1,69 @@
+import React, {useState, useContext, useCallback} from 'react';
+
+import Menu from '@material-ui/core/Menu';
+import MenuItem from '@material-ui/core/MenuItem';
+import IconButton from '@material-ui/core/IconButton';
+import MoreVertIcon from '@material-ui/icons/MoreVert';
+
+import {VulcanContext} from '../../../contexts/vulcan_context';
+import {VulcanFigure, VulcanSession} from '../../../api_types';
+import {
+  setSessionAndFigure,
+  setWorkflow
+} from '../../../actions/vulcan_actions';
+
+export default function AdvancedSessionControls({
+  session,
+  figure
+}: {
+  figure: VulcanFigure;
+  session: VulcanSession;
+}) {
+  const [menuAnchor, setMenuAnchor] = useState(
+    null as HTMLButtonElement | null
+  );
+  const {dispatch, updateFigureDependencies} = useContext(VulcanContext);
+
+  const handleClose = () => {
+    setMenuAnchor(null);
+  };
+
+  const updateDependencies = useCallback(() => {
+    // Post an update request for the figure,
+    //   then set local workflow / session with
+    //   the new reference_figure_id and snapshot.
+    if (!figure.figure_id) return;
+
+    updateFigureDependencies(session.project_name, figure.figure_id).then(
+      (updatedFigure) => {
+        dispatch(setSessionAndFigure(updatedFigure));
+        if (updatedFigure.workflow_snapshot)
+          dispatch(
+            setWorkflow(updatedFigure.workflow_snapshot, session.project_name)
+          );
+        handleClose();
+      }
+    );
+  }, [session, figure, updateFigureDependencies, dispatch]);
+
+  return (
+    <>
+      <IconButton
+        onClick={(e) => {
+          e.stopPropagation();
+          setMenuAnchor(e.currentTarget);
+        }}
+      >
+        <MoreVertIcon />
+      </IconButton>
+      <Menu
```

Comment:
Currently updating a figure's dependencies (SHA and workflow-snapshot) is a distinct action. I could also imagine this as a check-box on the "Save" button, so you could update at the same time you save the figure. Not sure which is preferable, since I tend to view updating dependencies as something you would rarely do, so have left it as separate for now.